DHCPv6: Add support of DUID

cathay4t opened the pull request:
Add `dhcp-duid` to `InterfaceIpv6` per RFC 8415:

    Clients and servers MUST treat DUIDs as opaque values and MUST only
    compare DUIDs for equality.  Clients and servers SHOULD NOT in any

Introduced these DHCPv6 DUID types:
 * `Dhcpv6Duid::LinkLayerAddressPlusTime`: showing as 'llt'.
 * `Dhcpv6Duid::LinkLayerAddress`: showing as 'll'.
 * `Dhcpv6Duid::EnterpriseNumber`: showing as 'en'.
 * `Dhcpv6Duid::Uuid`: showing as 'uuid'.
 * `Dhcpv6Duid::Other(String)`: allow user to define arbitrary value for
   DUID in hex format or backend specific string.

Currently, NetworkManager only supports `llt`, `ll` and raw hex string.
Integration test case included.

ffmancera reviewed (changes requested):
It looks good to me, but I have 2 nits:
* In the commit message the sentence: `Clients and servers SHOULD NOT in any` seems incomplete.
* If NetworkManager only supports `llt`, `ll` and raw hex string, I think we should fail on validation if the user set `uuid` or `en` and specify that it is not supported by NetworkManager.

What do you think?

cathay4t committed:
DHCPv6: Add support of DUID

Add `dhcp-duid` to `InterfaceIpv6` per RFC 8415:

    Clients and servers MUST treat DUIDs as opaque values and MUST only
    compare DUIDs for equality.  Clients and servers SHOULD NOT in any
    other way interpret DUIDs.

Introduced these DHCPv6 DUID types:
 * `Dhcpv6Duid::LinkLayerAddressPlusTime`: showing as 'llt'.
 * `Dhcpv6Duid::LinkLayerAddress`: showing as 'll'.
 * `Dhcpv6Duid::EnterpriseNumber`: showing as 'en'.
 * `Dhcpv6Duid::Uuid`: showing as 'uuid'.
 * `Dhcpv6Duid::Other(String)`: allow user to define arbitrary value for DUID
   in hex format or backend specific string.

Currently, NetworkManager only supports `llt`, `ll` and raw hex string.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
> It looks good to me, but I have 2 nits:
> 
> * In the commit message the sentence: `Clients and servers SHOULD NOT in any` seems incomplete.

Added the missing part.

> * If NetworkManager only supports `llt`, `ll` and raw hex string, I think we should fail on validation if the user set `uuid` or `en` and specify that it is not supported by NetworkManager.

We are allowing backends to report any special mode they have and verify it.
Actually NetworkManager document indicate it support many other modes, we can trust user knowing the meaning of them.

> 
> What do you think?
